### PR TITLE
Malformed share blocks recover button

### LINKED ISSUE #131: Invalid shares passed to the recovery function
# Bug

## Description

It is possible to select a malformed share plus a quorum of valid shares such that both the valid and invalid shares are passed along to rusty-secrets for processing, which fails.

## Steps to Reproduce

From the Recover screen, load an invalid share file (e.g. Beautiful Duck.png).
Then, without removing the first file, load a quorum of valid shares (e.g. secret-shard-{1-5}.txt).

### Actual Behavior

The Recover button appears, but clicking it causes the invalid shares to be passed along to the recover function, which fails.

### Expected Behavior

The Recover button does not appear as long as there are invalid shares present.

### PR DESCRIPTION
**Pull request by abhn**
## Status

Ready for review

## Description of Changes

Fixes #131 

When a malformed file is present in the uploaded shares, the "recover" button doesn't appear and a text message requests the user to remove all malformed shares

## Testing

- On the recover secret page, for the first secret share, add a random text file random123.txt
- Next, select the valid (N) shares that yield the secret

**Behavior prior to bug fix:** Recover button appears and upon clicking it, an error is thrown

**Behavior after bug fix:** Recover button is hidden and a message saying "**Remove malformed shares to continue.**" prompts the user to remove any malformed shares. After the malformed shares are removed, recover button is shown as usual.

### CODE REVIEW

**Comment by abhn** [comment]
@conorsch Any feedback on this one appreciated.

One thing I'm not sure is if package-lock.json should've been added to this. I did, but looking at other PRs, I think I shouldn't have. Let me know if it should be removed.

**Comment by liliakai** [comment]
Hi @abhn, thanks for your PR! I'm at an Electron conference all week, but I'll have time to take a closer look at this change next week.

In answer to your question, yes, please omit the changes to package-lock.json. The fact that it changed indicates that we have some outstanding dependency updates to dial in, but that should be a separate change.

**Comment by abhn** [comment]
Sure, thanks for the reply @liliakai I'll omit `package-lock.json`

File: src/components/Recover.js [hunk 1]
```diff
@@ -76,11 +77,11 @@ export default class Recover extends Component {
         </div>
       );
     }
-
+    
     return (
       <div className="container flex-column recover">
         {action}
-        {shouldDisplayStatus && <RecoverStatus quorum={quorum} shares={shares} />}
+        {shouldDisplayStatus ? <RecoverStatus quorum={quorum} shares={shares} /> : null}
```
[the remark this answers is not in the document]
Comment (abhn) [reply src/components/Recover.js]: `shouldDisplayStatus`, when 0 (that is, false), gets printed as literal `0` (like a rouge character below). This change fixes that.

**Pull request merged**